tests/shared/robot.spec.js

```diff
@@ -91,6 +91,97 @@ describe('Shared: Robot', function() {
+      m.chai.expect(robot.isMessage('{"command":"foo"}')).to.be.true;
+    });
+
+    it('should return false if message is an invalid object', function() {
+      m.chai.expect(robot.isMessage('{"command":\\foo"}')).to.be.false;
+    });
+
+    it('should return false if message is a string', function() {
+      m.chai.expect(robot.isMessage('foo')).to.be.false;
+    });
+
+    it('should return false if message is an empty string', function() {
+      m.chai.expect(robot.isMessage('')).to.be.false;
+    });
+
+    it('should return false if message is undefined', function() {
+      m.chai.expect(robot.isMessage(undefined)).to.be.false;
+    });
+
+    it('should return false if message is null', function() {
+      m.chai.expect(robot.isMessage(null)).to.be.false;
+    });
+
+    it('should return false if message is a positive integer string', function() {
+      m.chai.expect(robot.isMessage('5')).to.be.false;
+    });
+
+    it('should return false if message is a negative integer string', function() {
+      m.chai.expect(robot.isMessage('-3')).to.be.false;
+    });
+
+    it('should return false if message is a zero string', function() {
+      m.chai.expect(robot.isMessage('0')).to.be.false;
+    });
+
+    it('should return false if message is a positive float string', function() {
+      m.chai.expect(robot.isMessage('5.3')).to.be.false;
+    });
+
+    it('should return false if message is a negative float string', function() {
+      m.chai.expect(robot.isMessage('-2.1')).to.be.false;
+    });
+
+    it('should return false if message is a positive integer', function() {
+      m.chai.expect(robot.isMessage(5)).to.be.false;
+    });
+
+    it('should return false if message is a negative integer', function() {
+      m.chai.expect(robot.isMessage(-3)).to.be.false;
+    });
+
+    it('should return false if message is zero', function() {
+      m.chai.expect(robot.isMessage(0)).to.be.false;
+    });
+
+    it('should return false if message is a positive float', function() {
+      m.chai.expect(robot.isMessage(5.3)).to.be.false;
+    });
+
+    it('should return false if message is a negative float', function() {
+      m.chai.expect(robot.isMessage(-2.1)).to.be.false;
+    });
+
+    it('should return false if message is an array', function() {
+      m.chai.expect(robot.isMessage([ 'foo' ])).to.be.false;
+    });
+
+    it('should return false if message is an array string', function() {
+      m.chai.expect(robot.isMessage('["foo"]')).to.be.false;
+    });
+
+    it('should return true for a message built with .buildMessage()', function() {
+      m.chai.expect(robot.isMessage(robot.buildMessage('foo', {
+        message: 'bar'
+      }))).to.be.true;
+    });
+
+    it('should return true for a message built with .buildErrorMessage()', function() {
+      const error = new Error('foo');
+      m.chai.expect(robot.isMessage(robot.buildErrorMessage(error))).to.be.true;
+    });
+
```

Comment:
As strings can be JSON encoded as well (`JSON.stringify('asd')` ->` '"asd"'`), it might be good to add that case as well